src/ploomber/dag/dag.py

```diff
@@ -644,6 +645,12 @@ def _run_on_render(self):
             self._logger.debug('No on_render hook for dag '
                                '"%s", skipping', self.name)
 
+    def _deepcopy_safe(self):
+        try:
+            return deepcopy(self)
+        except Exception:
+            raise Exception("An error occurred while copying DAG object")
```

Comment:
please change this for a `RuntimeError` and add it as a chained exception (`raise RuntimeError(...) from e`)